extensions/amp-smartlinks/0.1/amp-smartlinks.js

```diff
@@ -0,0 +1,180 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Services} from '../../../src/services';
+
+import {LinkRewriterManager} from './link-rewriter/link-rewriter-manager';
+import {Linkmate} from './linkmate';
+import {NRTV_CONFIG_ENDPOINT,
+  PAGE_IMPRESSION_ENDPOINT,
+  SMARTLINKS_REWRITER_ID} from './constants';
+import {getConfigOptions} from './linkmate-options';
+
+
+export class AmpSmartlinks extends AMP.BaseElement {
+  /**
+   * @param {!AmpElement} element
+   */
+  constructor(element) {
+    super(element);
+
+    /** @private {?../../../src/service/xhr-impl.Xhr} */
+    this.xhr_ = null;
+
+    /** @private {?../../../src/service/ampdoc-impl.AmpDoc} */
+    this.ampDoc_ = null;
+
+    /** @private {?./link-rewriter/link-rewriter-manager.LinkRewriterManager} */
+    this.linkRewriterService_ = null;
+
+    /** @private {?./link-rewriter/link-rewriter.LinkRewriter} */
+    this.smartLinkRewriter_ = null;
+
+    /** @private {?../../../src/service/viewer-impl.Viewer} */
+    this.viewer_ = null;
+
+    /** @private {?./linkmate.Linkmate} */
+    this.linkmate_ = null;
+
+    /** @private {?string} */
+    this.referrer_ = null;
+  }
+
+  /** @override */
+  buildCallback() {
+    this.xhr_ = Services.xhrFor(this.win);
+    this.ampDoc_ = this.getAmpDoc();
+
+    this.viewer_ = Services.viewerForDoc(this.ampDoc_);
+
+    this.linkmateOptions_ = getConfigOptions(this.element);
+    this.linkmateConfig_ = this.getLinkmateOptions_();
+
+    this.linkRewriterService_ = new LinkRewriterManager(this.ampDoc_);
+
+    return this.ampDoc_.whenBodyAvailable()
+        .then(() => this.viewer_.getReferrerUrl())
+        .then(referrer => {
+          this.referrer_ = referrer;
+          this.runSmartlinks_();
+        });
+  }
+
+  /**
+   * @return {!JsonObject}
+   * @private
+   */
+  getLinkmateOptions_() {
+    const fetchUrl = NRTV_CONFIG_ENDPOINT.replace('.nrtv_slug.',
+        this.linkmateOptions_.nrtvSlug);
+
+    return this.xhr_.fetchJson(fetchUrl, {
+      method: 'GET',
+      ampCors: false,
+    })
+        .then(res => res.json())
```

Comment:
A bit weird indentation
```
}).then(res => res.json())
.then(res => {
```

Reply:
I just change indentation until the eslint errors go away. Can try this.